### ISSUE TITLE
Make all input arguments auto-discoverable and mark as @AutoDiscoverable

### ISSUE THREAD

**Issue by droazen** [broadinstitute/gatk#300]
Currently, Feature arguments for each tool are discovered automatically through reflection via FeatureManager and initialized/added to a query pool. Perhaps for the sake of consistency we should perform the same kind of auto-discovery for all common kinds of input arguments (reads, reference, etc.)

**Comment by akiezun** [comment]
agreed, consistency above all. Can you add an annotation that will alert the code reader that magic autodiscovery happens (eg @AutoDiscoverable) - the definition of that annotation will be a good place to put the doc explaining the magic

**Comment by droazen** [comment]
Like the idea of an explicit `@AutoDiscoverable` annotation.

**Comment by akiezun** [comment]
alpha candidate? or too much work?

**Comment by lbergelson** [comment]
Push this off I think

**Comment by droazen** [comment]
Not sure we actually want to do this one -- unassigning, but will keep open for debate.
